subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -2813,31 +2863,42 @@ void parse_mode_args_to_params(const struct shell *sh, int argc,
 		switch (opt) {
 		case 's':
 			mode->mode |= WIFI_STA_MODE;
+			opt_num++;
 			break;
 		case 'm':
 			mode->mode |= WIFI_MONITOR_MODE;
+			opt_num++;
 			break;
 		case 'a':
 			mode->mode |= WIFI_AP_MODE;
+			opt_num++;
 			break;
 		case 'k':
 			mode->mode |= WIFI_SOFTAP_MODE;
+			opt_num++;
 			break;
 		case 'g':
 			mode->oper = WIFI_MGMT_GET;
+			opt_num++;
 			break;
 		case 'i':
 			mode->if_index = (uint8_t)atoi(state->optarg);
+			/* Don't count iface as it's common for both get and set */
```

Comment:
We should fix also the atoi() here at some point (in separate commit or PR). In fact all the atoi calls in this file should be converted to `shell_strtol()` and the return values should be properly checked.

Reply:
Will raise a new PR